src/server/pps/server/worker_rc.go

```diff
@@ -466,10 +481,18 @@ func (a *apiServer) getWorkerOptions(ptr *pps.EtcdPipelineInfo, pipelineInfo *pp
 			}
 		}
 	}
+	var s3GatewayPort int32
+	if ppsutil.ContainsS3Inputs(pipelineInfo.Input) || pipelineInfo.S3Out {
+		// TODO(msteffen) do we need a separate option for the sidecar port?
+		// env.S3GatewayPort was originally added for main pachd.
```

Comment:
I think not, we'll likely want to randomize the port the s3g sidecar instance

Reply:
Comment removed